tests/integration/cgroup_test.go

```diff
@@ -0,0 +1,196 @@
+package integration
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	pb "github.com/mikewurtz/taskman/gen/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func parseCPUStat(t *testing.T, content string) map[string]uint64 {
+	t.Helper()
+	stats := make(map[string]uint64)
+	for _, line := range strings.Split(content, "\n") {
+		if line == "" {
+			continue
+		}
+		parts := strings.Fields(line)
+		if len(parts) != 2 {
+			t.Fatalf("invalid cpu.stat line: %q", line)
+		}
+		val, err := strconv.ParseUint(parts[1], 10, 64)
+		if err != nil {
+			t.Fatalf("invalid value in cpu.stat line: %q: %v", line, err)
+		}
+		stats[parts[0]] = val
+	}
+	return stats
+}
+
+func TestIntegration_StartTaskOOMKilledPerl(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "perl",
+		Args:    []string{"-e", "my $x = \"A\" x (128 * 1024 * 1024); sleep 5;"},
+	})
+	assert.NotNil(t, resp)
+	assert.NotEmpty(t, resp.TaskId)
+	require.NoError(t, err)
+	sts, ok := status.FromError(err)
+	require.True(t, ok)
+	assert.Equal(t, codes.OK, sts.Code())
+
+	var statusResp *pb.TaskStatusResponse
+	require.Eventually(t, func() bool {
+		statusResp, err = client.GetTaskStatus(ctx, &pb.TaskStatusRequest{
+			TaskId: resp.TaskId,
+		})
+		// wait for the task to exit with and error
+		return err == nil && statusResp != nil && statusResp.Status == pb.JobStatus_JOB_STATUS_SIGNALED
+	}, 3*time.Second, pollInterval, "expected task to exit with error")
+
+	assert.Equal(t, pb.JobStatus_JOB_STATUS_SIGNALED, statusResp.Status)
+	assert.Equal(t, "oom", statusResp.TerminationSource)
+	assert.Equal(t, syscall.SIGKILL.String(), statusResp.TerminationSignal)
+	assert.NotEmpty(t, statusResp.ProcessId)
+	assert.NotNil(t, statusResp.EndTime)
+	assert.NotNil(t, statusResp.StartTime)
+	assert.Equal(t, resp.TaskId, statusResp.TaskId)
+	assert.Nil(t, statusResp.ExitCode)
+}
+
+func TestIntegration_StartTaskIOThrottled(t *testing.T) {
+	t.Parallel()
+
+	client := createTestClient(t, "client001")
+
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+
+	resp, err := client.StartTask(ctx, &pb.StartTaskRequest{
+		Command: "dd",
+		Args: []string{
+			"if=/dev/sda",
```

Comment:
Additional prereq maybe worth noting in the README (or maybe auto detecting) - this will fail if the system does not have a `/dev/sda`.

Reply:
Good idea, I will get the readme updated.